platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/BitmapImageConversionTask.kt

```diff
@@ -0,0 +1,31 @@
+package org.maplibre.android.maps
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import org.maplibre.android.maps.Style.Builder.ImageWrapper
+import java.lang.ref.WeakReference
+
+internal class BitmapImageConversionTask(
+	nativeMap: NativeMap,
+) {
+	private val nativeMap = WeakReference(nativeMap)
+	private val job: Job = SupervisorJob(null)
+	private val scope = CoroutineScope(Dispatchers.IO + job)
+
+	fun execute(vararg params: ImageWrapper) {
+		scope.launch {
+			val images = params.map { Style.toImage(it) }
+			val nativeMap = nativeMap.get()
+			if (nativeMap != null && !nativeMap.isDestroyed) {
+				nativeMap.addImages(images.toTypedArray())
+			}
+		}
+	}
+
+	fun cancel() {
+		job.cancel()
+	}
```

Comment:
`launch` returns a `Job`. You could also return it in `execute`.

In any case, (separate issue), here you cancel the `SupervisorJob`, i.e. the parent job. After the `SupervisorJob` is cancelled, IIRC no new jobs can be launched in that job. IIRC subsequent calls to `execute` would then just immediately cancel.
That would be fine if the `BitmapImageConversionTask` is a single-use object, but then, why does it need a `SupervisorJob`? In any case, needs documentation.

Reply:
The `execute` call could potentially be launched several times.  The intent was that the `SupervisorJob` would cancel all of them.  Note that `BitmapImageConversionTask.cancel` is only called when the `Style` is cleared.

I'll add a comment to clarify that.